src/internet-computer-programming-concepts/basic-memory-persistence/stable-variables.md

```diff
@@ -34,27 +37,58 @@ Our actor has a mutable variable `count` that is declared `stable`. It's initial
+
+Stable variables can only be declared inside [actors](/internet-computer-programming-concepts/actors.html). Stable variables always have `private` *visibility*. 
+
+The following *types* for *[immutable](common-programming-concepts/variables.html) or [mutable](/common-programming-concepts/mutability.html)* variables in actors (in addition to all [shared types](/internet-computer-programming-concepts/async-data/shared-types.html)) could be declared stable.
+
+### Stable Mutable Array
+Immutable or mutable variables of [*mutable array type*](/common-programming-concepts/types/mutable-arrays.html) could be declared stable:
+```motoko
+{{#include _mo/stable-var1.mo:a}}
+```
 
-<!-- ## Stable var types vs shared types
-Recall that [shared types](/internet-computer-programming-concepts/async-data/shared-types.html) are always immutable. On the other hand, stable variables are always mutable. A subtle fact is that the *values* of *mutable stable variables* are restricted to values of shared types only.
+*Immutable variable* `a1` can be stable because `[var Nat]` is a mutable array type. *Mutable variable* `a2` can be stable because `[var Text]` is of mutable array type.   
 
-### Non stable types
-Suppose we have an [object](/common-programming-concepts/objects-and-classes/objects.html) like:
+### Stable Records with Mutable Fields
+Immutable or mutable variables of [*records with mutable fields*](/common-programming-concepts/types/records.html) could be declared stable:
 ```motoko
-{{#include _mo/stable-variables3.mo:a}}
+{{#include _mo/stable-var1.mo:b}}
 ```
 
-We define a type of an object with a public field that is mutable. We assign an instance of that object to an immutable variable named `myObject`. We could now mutate the state of our public variable.
+*Immutable variable* `r1` and *mutable variable* `r2` can be stable because they have a value of a *record with a mutable field*. 
+
+### Stable Objects with Mutable Variables
+Immutable or mutable variables of [*objects with (private or public) mutable variables*](/common-programming-concepts/objects-and-classes/objects.html) could be declared stable:
+```motoko
+{{#include _mo/stable-var1.mo:c}}
+```
+
+*Immutable variable* `o1` and *mutable variable* `o2` can be stable because they have a value of an *object with private and public mutable variables*. 
+
+### Stable Objects with Private Functions
+Immutable or mutable variables of [*objects with private functions*](/common-programming-concepts/objects-and-classes/objects.html) could be declared stable:
+```motoko
+{{#include _mo/stable-var1.mo:d}}
+```
```

Comment:
Objects with only private functions aren't really objects. Maybe turn this into a negative example instead by making the functions public, in which case the declaration will be rejected because it can't be stable.